### PR TITLE
Partial parsing and remodel of GameModel

### PR DESCRIPTION
**Pull request by roxannecvl**
This PR will give to the main the parsing of some of the hints, it also changes GameModel such that now there is no more bug with promises. This PR has to be accepted in order to create a v1 game for the Mid-project review

### CODE REVIEW

**Review by martinctl — APPROVED** [review]

File: model/GameModel.ts [hunk 1]
```diff
@@ -119,12 +121,73 @@ export class GameModel {
...
     set blur(value: number) {
         if (value < 0 || value > 7) {
             throw new Error("Blur must be between 0 and 7");
         }
         this._blur = value;
     }
+
+    setIntroPromiseState(prmState : PromiseState){
+        this.introPromiseState = prmState;
+    }
```
Comment (JEANPRBT) [inline model/GameModel.ts:137]: Remove this method.

File: model/GameModel.ts [hunk 2]
```diff
@@ -1,67 +1,69 @@
...
     private _curHintLevel: number = 1;
     private _nbGuesses = 0;
     private _curGuess : string = '';
     private _prevGuesses : string[] = [];
-    private _promiseState: any = {};
+    public introPromiseState: PromiseState = { data: null, promise: null, error: null };
+    public imagePromiseState: PromiseState = { data: null, promise: null, error: null };
+    public infoPromiseState: PromiseState = { data: null, promise: null, error: null };
     private _end: boolean = false;
     private _win: boolean = false;
```
Comment (JEANPRBT) [inline model/GameModel.ts:30]: Try to initialize these values in the contructor.

File: model/GameModel.ts [hunk 3]
```diff
@@ -1,67 +1,69 @@
-import { resolvePromise } from "./resolvePromise";
-import { HintList, Hint } from "~/model/HintList";
+import { resolvePromise} from "~/model/resolvePromise"
+import type {PromiseState } from "~/model/resolvePromise"
+import {Hint, HintList} from "~/model/HintList"
+import {fetchIntro, fetchImageUrl, fetchInfoBox} from "~/api/wikipediaSource"
```
Comment (JEANPRBT) [inline model/GameModel.ts:4]: Try to be consistent with the spaces you use when importing.

File: package.json [hunk 4]
```diff
@@ -16,10 +16,10 @@
     "firebase-admin": "^11.11.1",
     "firebase-functions": "^4.8.0",
     "fuse.js": "^6.4.6",
-    "infobox-parser": "^3.6.4",
     "nuxt": "^3.10.3",
     "nuxt-vuefire": "^1.0.2",
     "tailwindcss": "^3.4.1",
+    "infobox-parser": "^3.6.4",
```
Comment (JEANPRBT) [inline package.json:22]: Remove this dependency.

File: pages/example.vue [hunk 5]
```diff
@@ -9,6 +9,8 @@ import InfoboxView
 import {
   GameModel
 } from "~/model/GameModel";
+import {celebrities} from "~/model/CelebrityList";
+import {Utils} from "~/utilities/Utils";
```
Comment (JEANPRBT) [inline pages/example.vue:13]: See above.

**Review by JEANPRBT — APPROVED** [review]

**Pull request merged**